src/code-template/fetch-all-code-templates.ts

```diff
@@ -0,0 +1,107 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import prisma from "@/prisma";
+import Joi from "joi";
+import { Prisma } from "@prisma/client";
+import { formatTags } from "../utils";
+import {
+  MAX_CHARS_CONTENT,
+  MAX_CHARS_TITLE_DESCRIPTION,
+  MAX_PAGE_SIZE,
+  MIN_PAGE_SIZE,
+  MIN_PAGES,
+  PAGE_SIZE,
+} from "../constants";
+
+type Error = {
+  error: string;
+};
+
+type CodeTemplateData = {
+  codeTemplates: any[];
+  total: number;
+  page: number;
+  pageSize: number;
+};
+
+const getCodeTemplatesSchema = Joi.object({
+  page: Joi.number().integer().min(MIN_PAGES).default(MIN_PAGES),
+  pageSize: Joi.number()
+    .integer()
+    .min(MIN_PAGE_SIZE)
+    .max(MAX_PAGE_SIZE)
+    .default(PAGE_SIZE),
+  tags: Joi.alternatives(
+    Joi.string().max(MAX_CHARS_TITLE_DESCRIPTION),
+    Joi.array().items(Joi.string().max(MAX_CHARS_TITLE_DESCRIPTION))
+  ).optional(),
+  title: Joi.string().max(MAX_CHARS_TITLE_DESCRIPTION).optional(),
+  code: Joi.string().max(MAX_CHARS_CONTENT).optional(),
```

Comment:
while it is good to add this filter, the endpoint currently returns a 400 BAD REQUEST when for example tags are more than `MAX_CHAR_TITLE_DESCRIPTION` in length.

It is not really a bad request, maybe it would be when creating code-templates though.

You could simply leave out the check for max length here if you like.